backend/planty/dashboard/views.py

```diff
@@ -290,3 +295,234 @@ def post(self, request: Request):
         plant.save()
 
         return Response(status=status.HTTP_200_OK)
+
+
+class InstructionsView(APIView):
+    def get(self, request: Request):
+        user: User = request.user
+        my_instructions = Instruction.objects.filter(user=user, public='False')
+
+        my_instructions_json = []
+        for instruction in my_instructions:
+            instruction_json = {
+                'id': str(instruction.id),
+                'name' : instruction.name,
+                'species': instruction.species,
+
+                'watering': instruction.watering,
+                'insolation': instruction.insolation,
+                'fertilizing': instruction.fertilizing,
+            }
+            my_instructions_json.append(instruction_json)
+
+
+        if request.method == 'GET' and 'limit_popular' in request.GET:
+            N = request.GET['limit_popular']
+        else:
+            N = 10
+
+        suggested = Instruction.objects.filter(public='True')
+        # returns best N suggested instruction
+        desc_suggested_instruction = suggested.order_by('num_selected').reverse()[:N]
+
+
+        desc_suggested_instructions_json = []
+        for instruction in desc_suggested_instruction:
+            instruction_json = {
+                'id': str(instruction.id),
+                'name': instruction.name,
+                'species': instruction.species,
+
+                'watering': instruction.watering,
+                'insolation': instruction.insolation,
+                'fertilizing': instruction.fertilizing,
+
+                'num_selected': instruction.num_selected
+            }
+            desc_suggested_instruction.append(instruction_json)
+
+        instructions_json = {
+            my: my_instructions_json,
```

Comment:
mam mieszane uczucia co do łączenia instrukcji użytkownika i sugerowanych w jeden endpoint

ja bym to jednak rozdzielił na 2 osobne endpointy, bo np. przy wyświetlaniu listy instrukcji użytkownika nie potrzebujemy sugerowanych

Reply:
w `instructions/` wszystkie metody HTTP oprócz GET'a korzystają w zasadzie tylko z instrukcji własnych użytkownika
mogę stworzyć nową klasę `PopularInstructions` i rozdzielić ten GET na 2 endpointy: `instructions/` tylko z własnymi instrukcjami i `instructions/popular` tylko z najpopularniejszymi instrukcjami
takie coś będzie ok?

Reply:
Może tak być

Reply:
GET'y rozdzielone